share/getters/shrex.go

```diff
@@ -145,6 +149,8 @@ func (sg *ShrexGetter) GetSharesByNamespace(
 		case errors.Is(getErr, p2p.ErrNotFound):
 			getErr = share.ErrNotFound
 			setStatus(peers.ResultCooldownPeer)
+		case errors.Is(err, routing.ErrNotFound):
```

Comment:
can you extract these switches into some common handling logic in shrex.go?

Reply:
There are a lot of common logic in shrex protocols, I would rather do abstraction in different PR.